Can't compile NewArchitecture branch on Linux.

I'm running Arch based Manjaro Xfce 64 bit. I already have the latest wxWidgets 3.0.2 installed and working. All other dependencies appear to install as well. I follow the Linux build instructions:  

_Build TrenchBroom_
 Open a terminal and change into the directory where you unpacked the TrenchBroom sources
Create a new directory, e.g. "build", and change into it.
Run the following two commands
cmake .. -DCMAKE_BUILD_TYPE=Release
cmake --build
*You can replace "Release" with "Debug" if you want to create a debug build. *
##### cmake .. -DCMAKE_BUILD_TYPE=Release RESULTS:

_[retro@ToolBox build]$ cmake .. -DCMAKE_BUILD_TYPE=Release
-- The C compiler identification is GNU 4.9.1
-- The CXX compiler identification is GNU 4.9.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Build type Release
-- Using static wxWidgets library
-- Found wxWidgets: TRUE  
-- Found FREETYPE: /usr/lib64/libfreetype.so  
-- Found FreeImage: /usr/lib64/libfreeimage.so  
-- Compiler is GNU
-- Found Git: /usr/bin/git (found version "2.1.2") 
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connectapprecuated
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib64/libX11.so
-- Found OpenGL: /usr/lib64/libGL.so  
fatal: Not a git repository (or any of the parent directories): .git
-- Configuring done
-- Generating done
-- Build files have been written to: /home/retro/compileme/TrenchBroom-NewArchitecture/build
[retro@ToolBox build]$_
##### Followed up by: cmake --build which result in nothing but the manual page help:

_[retro@ToolBox build]$ cmake --build
Usage: cmake --build <dir> [options] [-- [native-options]]
Options:
  <dir>          = Project binary directory to be built.
  --target <tgt> = Build <tgt> instead of default targets.
  --config <cfg> = For multi-configuration tools, choose <cfg>.
  --clean-first  = Build target 'clean' first, then build.
                   (To clean only, use --target 'clean'.)
  --use-stderr   = Ignored.  Behavior is default in CMake >= 3.0.
  --             = Pass remaining options to the native tool.
[retro@ToolBox build]$_

I'm I doing something wrong? Is there something obvious I am missing?

FYI I've also installed Ubuntu 14 LTS in a virtual environment for testing, then all the TB 2 dependencies and the exact same results as above.

Following someones help I attempt again from fresh with:

_mkdir build && cd build
cmake ..
make_

Same results up to "cmake .." BUT when I ran "make" it actually started compiling for a few lines before it met with a missing file  #include "Model/ObjectParent.h"  needed to build.

_[retro@ToolBox build]$ make
Scanning dependencies of target version
fatal: Not a git repository (or any of the parent directories): .git
[  0%] Built target version
Scanning dependencies of target glew
[  0%] Building C object CMakeFiles/glew.dir/lib/src/glew/glew.c.o
Linking C static library libglew.a
[  0%] Built target glew
Scanning dependencies of target common
[  0%] Building CXX object CMakeFiles/common.dir/common/src/Hit.cpp.o
[  0%] Building CXX object CMakeFiles/common.dir/common/src/MathUtils.cpp.o
[  1%] Building CXX object CMakeFiles/common.dir/common/src/Renderer/MapRenderer.cpp.o
In file included from /home/retro/compileme/TrenchBroom-NewArchitecture/common/src/LayerObserver.h:24:0,
                 from /home/retro/compileme/TrenchBroom-NewArchitecture/common/src/Renderer/MapRenderer.h:23,
                 from /home/retro/compileme/TrenchBroom-NewArchitecture/common/src/Renderer/MapRenderer.cpp:20:
/home/retro/compileme/TrenchBroom-NewArchitecture/common/src/Model/Layer.h:27:32: fatal error: Model/ObjectParent.h: No such file or directory
 #include "Model/ObjectParent.h"
                                ^
compilation terminated._
##### And indeed when checking where "ObjectParent.h" should be, where it's expected to be, its not there.

Any help appreciated.

--- a/common/src/IO/Hexen2MapWriter.cpp
+++ b/common/src/IO/Hexen2MapWriter.cpp
@@ -19,7 +19,7 @@
 
 #include "Hexen2MapWriter.h"
 
-#include "Model/Brushface.h"
+#include "Model/BrushFace.h"
 
 namespace TrenchBroom {
     namespace IO {

--- a/common/src/IO/StandardMapWriter.cpp
+++ b/common/src/IO/StandardMapWriter.cpp
@@ -20,7 +20,7 @@
 #include "StandardMapWriter.h"
 
 #include "StringUtils.h"
-#include "Model/Brushface.h"
+#include "Model/BrushFace.h"
 
 namespace TrenchBroom {
     namespace IO {

--- a/common/src/IO/ValveMapWriter.cpp
+++ b/common/src/IO/ValveMapWriter.cpp
@@ -19,7 +19,7 @@
 
 #include "ValveMapWriter.h"
 
-#include "Model/Brushface.h"
+#include "Model/BrushFace.h"
 
 namespace TrenchBroom {
     namespace IO {

